src/App.tsx

```diff
@@ -15,48 +16,116 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  BY_NONE = 0,
+  BY_ALPHABET,
+  BY_LENGTH,
+}
+
+type SortQuery = {
+  sortBy: SortType;
+  isSortReversed: boolean;
+};
+
+function getPreparedGoods(
+  goods: string[],
+  { sortBy, isSortReversed }: SortQuery,
+) {
+  const preparedGoods = [...goods].sort((a, b) => {
+    switch (sortBy) {
+      case SortType.BY_ALPHABET:
+        return a.localeCompare(b);
+
+      case SortType.BY_LENGTH:
+        return (a.length - b.length);
+
+      default:
+        return 0;
+    }
+  });
+
+  return isSortReversed ? preparedGoods.reverse() : preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(SortType.BY_NONE);
+  const [isSortReversed, setIsSortReversed] = useState(false);
+  const handleSortByAlphabet = () => {
+    setSortBy(sortBy === SortType.BY_ALPHABET
+      ? SortType.BY_NONE
+      : SortType.BY_ALPHABET);
+  };
+
+  const handleSortByLength = () => {
+    setSortBy(sortBy === SortType.BY_LENGTH
+      ? SortType.BY_NONE
+      : SortType.BY_LENGTH);
+  };
+
+  const handleSortReversed = () => {
+    setIsSortReversed(prev => !prev);
+  };
+
+  const handleSortReset = () => {
+    setSortBy(SortType.BY_NONE);
+    setIsSortReversed(false);
+  };
+
+  const visibleGoods = getPreparedGoods(goodsFromServer,
+    { sortBy, isSortReversed });
+
```

Comment:
```suggestion
  const visibleGoods = getPreparedGoods(goodsFromServer,
    { sortBy, isSortReversed }
  );

```

Reply:
![image](https://github.com/mate-academy/react_list-of-goods-hooks/assets/49778626/989572ec-316d-418c-a0a9-522f52be4a5c)

Reply:
I see, you can disregard the previous option.